ovos_audio/service.py

```diff
@@ -415,31 +420,52 @@ def _resolve_sound_uri(uri: str):
             raise FileNotFoundError(f"{audio_file} does not exist")
         return audio_file
 
+    @staticmethod
+    def _path_from_hexdata(hex_audio, audio_ext=None):
+        fname = md5(hex_audio.encode("utf-8")).hexdigest()
+        bindata = binascii.unhexlify(hex_audio)
+        if not audio_ext:
+            LOG.warning("audio extension not sent, assuming wav")
+            audio_ext = "wav"
+
+        audio_file = f"{gettempdir()}/{fname}.{audio_ext}"
+        with open(audio_file, "wb") as f:
+            f.write(bindata)
+        return audio_file
+
     def handle_queue_audio(self, message):
         """ Queue a sound file to play in speech thread
          ensures it doesnt play over TTS """
         if not validate_message_context(message):
             LOG.debug("ignoring playback, message is not from a native source")
             return
         viseme = message.data.get("viseme")
-        audio_ext = message.data.get("audio_ext")  # unused ?
         audio_file = message.data.get("uri") or \
                      message.data.get("filename")  # backwards compat
+        hex_audio = message.data.get("binary_data")
+        audio_ext = message.data.get("audio_ext")
+        if hex_audio:
+            audio_file = self._path_from_hexdata(hex_audio, audio_ext)
+
         if not audio_file:
             raise ValueError(f"'uri' missing from message.data: {message.data}")
         audio_file = self._resolve_sound_uri(audio_file)
         audio_ext = audio_ext or audio_file.split(".")[-1]
         listen = message.data.get("listen", False)
 
         sess_id = SessionManager.get(message).session_id
-        TTS.queue.put((audio_ext, str(audio_file), viseme, sess_id, listen, message))
+        TTS.queue.put((str(audio_file), viseme, sess_id, listen, message))
 
     def handle_instant_play(self, message):
         """ play a sound file immediately (may play over TTS) """
         if not validate_message_context(message):
             LOG.debug("ignoring playback, message is not from a native source")
             return
         audio_file = message.data.get("uri")
+        hex_audio = message.data.get("binary_data")
+        audio_ext = message.data.get("audio_ext")
+        if hex_audio:
+            audio_file = self._path_from_hexdata(hex_audio, audio_ext)
         if not audio_file:
             raise ValueError(f"'uri' missing from message.data: {message.data}")
```

Comment:
This should be updated to something like "no `uri` or `binary_data` in message.data"